### ISSUE TITLE
Indexing owningId and releatedId in mongo _Join documents

### ISSUE THREAD

**Issue by JasCodes** [parse-community/parse-server#7832]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

Indexes are not automatically added when _Join: documents are created

### Feature / Enhancement Description

By adding index for owningId and relatedId the query performance of mongo should improve. @davimacedo @mtrezza  Now unless I am mistaken and it should not be required, let me know?
<img width="1628" alt="Untitled" src="https://user-images.githubusercontent.com/1231593/156763723-429b15c8-980d-4004-89dd-95f664808aaf.png">

### Alternatives / Workarounds

Manually adding indexes

**Comment by davimacedo** [comment]
I believe these indexes are not currently automatic created. I also believe that it would be a good addition.

**Comment by JasCodes** [comment]
@mtrezza  @davimacedo Any idea where the collection _Join is being created?

**Comment by mtrezza** [comment]
Did you search the code base yet?

**Comment by JasCodes** [comment]
Yeah, bit confused. Still trying to get my head around codebase. Any hint for this specific instance?

**Comment by mtrezza** [comment]
Did you do a simple text search for "_Join"?

It yields the following from which you can backtrack the callers.

https://github.com/parse-community/parse-server/blob/0c3feaaa1751964c0db89f25674935c3354b1538/src/Controllers/DatabaseController.js#L296

**Comment by Moumouls** [comment]
@jascodes i seen this also recently, adding indexes will for sure improve performance.

And yes it's hard to find because... _Join classes are not created by code on Parse Server. These classes are created on the fly by mongo when you save the first Relation join object.
Mongo docs:
> If a collection does not exist, MongoDB creates the collection when you first store data for that collection.

Relation add is triggered here

```js
// src/Controllers/DatabaseController.js
  addRelation(key: string, fromClassName: string, fromId: string, toId: string) {
    const doc = {
      relatedId: toId,
      owningId: fromId,
    };
    return this.adapter.upsertOneObject(
      `_Join:${key}:${fromClassName}`,
      relationSchema,
      doc,
      doc,
      this._transactionalSession
    );
  }
```

In term of scope since _Join class is specific to the mongo adapter. I can suggest to add a simple key "joinClassIndexesCreated = {}" on the Mongo Adapter. Then after the `upsertOneObject` if you check

```js
if(this.joinClassIndexesCreated[`_Join:${key}:${fromClassName}`]) return result
// here we are not sure to have indexes created for this Join call
// first of all since it's an in memory strategy, fetch indexes of the _Join collection (use mongo directly) 
// then if needed, use also mongo directly to create indexes: collection._mongoCollection.createIndex 
```

Feel free to switch the addRelation fn to async/await
